File: snobot_sim_java/src/main/java/com/snobot/simulator/module_wrapper/wpi/WpiSolenoidWrapper.java
package com.snobot.simulator.module_wrapper.wpi;

import com.snobot.simulator.module_wrapper.ASensorWrapper;
import com.snobot.simulator.module_wrapper.interfaces.ISolenoidWrapper;

public class WpiSolenoidWrapper extends ASensorWrapper implements ISolenoidWrapper
{
    private boolean mState;

    public WpiSolenoidWrapper(int aIndex)
    {
        super("Solenoid " + aIndex);
        mState = false;
    }

    @Override
    public boolean get()
    {
        return mState;
    }

    @Override
    public void set(boolean aState)
    {
        mState = aState;
    }
}

File: snobot_sim/src/main/native/cpp/SnobotSim/ModuleWrapper/WpiWrappers/WpiSolenoidWrapper.cpp
/*
 * WpiSolenoidWrapper.cpp
 *
 *  Created on: May 3, 2017
 *      Author: PJ
 */

#include "SnobotSim/ModuleWrapper/WpiWrappers/WpiSolenoidWrapper.h"

#include "SnobotSim/PortUnwrapper.h"
#include "mockdata/PCMData.h"

const std::string WpiSolenoidWrapper::TYPE = "com.snobot.simulator.module_wrapper.wpi.WpiSolenoidWrapper";

WpiSolenoidWrapper::WpiSolenoidWrapper(int aPort) :
        AModuleWrapper("Solenoid " + std::to_string(UnwrapPort(aPort))),
        mModule(aPort / 8),
        mChannel(aPort % 8)
{
}

WpiSolenoidWrapper::~WpiSolenoidWrapper()
{
}

void WpiSolenoidWrapper::SetState(bool aOn)
{
    HALSIM_SetPCMSolenoidOutput(mModule, mChannel, aOn);
}

bool WpiSolenoidWrapper::GetState()
{
    return HALSIM_GetPCMSolenoidOutput(mModule, mChannel);
}
